java/core/src/java/org/apache/orc/impl/OrcTail.java

```diff
@@ -70,8 +70,11 @@ public long getFileModificationTime() {
 
   public OrcFile.WriterVersion getWriterVersion() {
     OrcProto.PostScript ps = fileTail.getPostscript();
+    OrcProto.Footer footer = fileTail.getFooter();
+    OrcFile.WriterImplementation writer =
+        OrcFile.WriterImplementation.from(footer.getWriter());
     return (ps.hasWriterVersion()
```

Comment:
I suggest we make `writerVersion` required for the new writer implementations, so you would just throw in that case...  This is similar to the reason we are starting with writer version 6